packages/node-config-provider/src/configLoader.ts

```diff
@@ -37,7 +37,15 @@ export const loadConfig = <T = string>(
   { environmentVariableSelector, configFileSelector, default: defaultValue }: LoadedConfigSelectors<T>,
   configuration: LocalConfigOptions = {}
 ): Provider<T> => {
-  const envOptions = configuration.signingName ? { signingName: configuration.signingName } : undefined;
+  const envOptions: EnvOptions = {};
+
+  if (configuration.signingName !== undefined) {
+    envOptions.signingName = configuration.signingName;
+  }
+  if (configuration.logger !== undefined) {
+    envOptions.logger = configuration.logger;
+  }
```

Comment:
since they're already optional, assign without check?

```ts
const { signingName, logger } = configuration;
const envOptions: EnvOptions = { signingName, logger };
```

Reply:
Done in [`05c748f` (#1584)](https://github.com/smithy-lang/smithy-typescript/pull/1584/commits/05c748f5f5cc77782afcf70c596b9d1df99a1e9b)